### PR TITLE
Make Archlinux ARM image lighter

### PR DESCRIPTION
**Pull request by janbuchar**

### CODE REVIEW

**Comment by janbuchar** [comment]
This is just the beginning though. We should call this image `archlinuxarm-dev` or something like that and have a separate image that doesn't include `gcc-libs` and the likes. The new image would be used for images that do not need to build stuff.

**Comment by petrbel** [comment]
@FloopCZ could you pls review this one? I don't think I understand the consequences enough.

Btw I think creating various images like `-dev` is a great idea. I'd use it happily in multiple projects. I am happy to set up all the automated builds etc once the dockerfiles are ready. @Teyras would you mind implementing them? Seems like you know what to do.

**Comment by FloopCZ** [comment]
Thank you @Teyras for this, great job!

I have followed up on your suggestions and split the image in two - `archlinuxarm` and `archlinuxarm-dev`. Could you please take a look at it? Moreover, would it be possible to recommend a similar "diet" for our x86 images? Thank you!

**Review by FloopCZ — APPROVED** [review]

**Comment by FloopCZ** [comment]
And by the way, the compressed image has just 140MB! (https://hub.docker.com/r/iterait/archlinuxarm/tags) FYI @petrbel

**Comment by petrbel** [comment]
woow, pretty cool!

File: dockerfiles/archlinuxarm/build.sh [hunk 1]
```diff
@@ -4,12 +4,21 @@
...
+# Build the base image.
+docker build -f archlinuxarm -t "iterait/archlinuxarm:${TODAY}" -t "iterait/archlinuxarm:latest" --squash .
+# Build the -dev version without context.
+docker build - -t "iterait/archlinuxarm-dev:${TODAY}" -t "iterait/archlinuxarm-dev:latest" < archlinuxarm-dev
+
+# Push the images to the repository.
+docker push iterait/archlinuxarm:latest
+docker push iterait/archlinuxarm:${TODAY}
+docker push iterait/archlinuxarm-dev:latest
+docker push iterait/archlinuxarm-dev:${TODAY}
```
Comment (petrbel) [inline dockerfiles/archlinuxarm/build.sh]: Do we actually need to push the images? It should be done automatically...

Comment (FloopCZ) [reply dockerfiles/archlinuxarm/build.sh]: However, the image needs to be built on ARM device. Therefore, we build and push it manually on one of our ARM devices whenever needed.

**Comment by janbuchar** [comment]
Before merging this, I'll add an ALPM hook that removes useless stuff (man pages and the likes) so that we don't have to remove it after each pacman/trizen command. We can also share this with the x86 image.

**Review by FloopCZ — COMMENTED** [review]
Hi @Teyras, thank you. I believe you forgot to `git add` the hook itself, though.

**Comment by janbuchar** [comment]
You're right again, dammit

**Review by FloopCZ — APPROVED** [review]
Everything looks good! Thanks!

**Comment by janbuchar** [comment]
Shall we merge this and proceed to trimming down the x86 image?

**Comment by FloopCZ** [comment]
Yes, please! :-)

**Pull request merged**